service/telemetry.go

```diff
@@ -50,6 +51,10 @@ const AddCollectorVersionTag = true
 const (
 	zapKeyTelemetryAddress = "address"
 	zapKeyTelemetryLevel   = "level"
+
+	// useOtelForInternalMetricsfeatureGateID is the feature gate ID that controls whether the collector uses open
+	// telemetry for internal metrics.
+	useOtelForInternalMetricsfeatureGateID = "telemetry.useOtelForInternalMetrics"
```

Comment:
not sure if camel case is the right choice here. @dashpole what is the k8s recommendation here?

Reply:
Looking at the [k8s feature gate](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) it seems like they follow the camel case convention. Seems like this is also the [suggestion](https://github.com/open-telemetry/opentelemetry-collector/blob/main/service/featuregate/README.md) of the original author for the feature gate in the repo also.

Reply:
camel-case is what other feature flags use, so we should go with that.  Kubernetes doesn't use `.` to categorize, but we already do that elsewhere (e.g. `receiver.otlp.newFeatureToAdd`).  This looks correct to me.

Reply:
@splunkericl @dashpole thanks, looks good.